### PR TITLE
Swift: Add some summary queries.

### PR DESCRIPTION
**Pull request by geoffw0**
I wrote these queries to assess the progress we're making identifying flow sources and sensitive expressions in real world Swift snapshots.  I intend to track these things fairly often in the near-mid future, so it would be convenient to have the queries in the public repo.  I don't think there's any harm in letting curious users see them as well.

It might be nice to have `LinesOfCode.ql` and `LinesOfUserCode.ql` summary queries as well as all (?) other languages do, but I don't think we extract the data for them yet.

### CODE REVIEW

File: swift/ql/src/queries/Summary/SummaryStats.ql [hunk 1]
```diff
@@ -0,0 +1,27 @@
+/**
+ * @name Summary statistics
+ * @description A table of summary statistics about a database. Includes
+ *              values that measure its size, and the numbers of certain
+ *              features interesting to analysis that have been found.
```
Comment (jketema) [inline swift/ql/src/queries/Summary/SummaryStats.ql:5]: I have a very hard time parsing "and the numbers of certain features interesting to analysis that have been found". What does "that have been found" refer to? "features" or "analysis"? Maybe just drop that part? I don't think that's quite sufficient for me to understand what you're trying to say, but it's a step.

**Review by jketema — COMMENTED** [review]
Mostly LGTM. Just one description I do not fully understand.

File: swift/ql/src/queries/Summary/SummaryStats.ql (hunk 1 above)
Comment (geoffw0) [reply swift/ql/src/queries/Summary/SummaryStats.ql]: You're right.  I've removed it.  Keep it simple.

**Comment by geoffw0** [comment]
In addition to your suggested change I've restricted the count of "Expressions" to ones with locations in code now.  It turns out there's quite a lot of locationless stuff that was exaggerating the size of smaller projects (and in turn making me wonder why we weren't finding many results, taint sources etc in them).

**Review by atorralba — DISMISSED** [review]
This LGTM. I guess summary queries don't need a docs review.

File: swift/ql/src/queries/Summary/SummaryStats.ql [hunk 2]
```diff
@@ -0,0 +1,25 @@
...
+ */
+
+import swift
+import codeql.swift.dataflow.FlowSources
+import codeql.swift.security.SensitiveExprs
+
+predicate statistic(string what, int value) {
+  what = "Files" and value = count(File f)
+  or
+  what = "Expressions" and value = count(Expr e | e.getFile().getName() != "")
```
Comment (jketema) [inline swift/ql/src/queries/Summary/SummaryStats.ql:16]: For C++ we have:
```
class UnknownLocation extends Location {
  UnknownLocation() { this.getFile().getAbsolutePath() = "" }
}
```
Doesn't that exist for Swift?

Comment (MathiasVP) [reply swift/ql/src/queries/Summary/SummaryStats.ql:16]: Indeed, Swift also has an `UnknownLocation` class: https://github.com/github/codeql/blob/30fc6acb19b5ddf8a23dd667d41b8750931bc643/swift/ql/lib/codeql/swift/elements/UnknownLocation.qll#L2.

Comment (geoffw0) [reply swift/ql/src/queries/Summary/SummaryStats.ql:16]: If it does, I should use it.  If it doesn't, I should add it...

**Comment by geoffw0** [comment]
> I guess summary queries don't need a docs review.

As far as I can tell summary queries don't usually get full `.qhelp` documentation, just an explanatory `@description`.  They're not really as visible as our main 'problem' queries so this seems appropriate to me.  Happy to add full docs, or get the `@description`s reviewed by the doc team if you would prefer.

File: swift/ql/src/queries/Summary/SummaryStats.ql (hunk 2 above)
Comment (geoffw0) [reply swift/ql/src/queries/Summary/SummaryStats.ql:16]: Thanks @MathiasVP !

**Comment by atorralba** [comment]
> Happy to add full docs, or get the @descriptions reviewed by the doc team if you would prefer.

No, let's keep it consistent with the rest of summary queries. Looks good as it is.

**Review by jketema — APPROVED** [review]

**Pull request merged**